hub-detect/src/main/resources/hub-detect-4-sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+# To override the default version key, specify your
+# own DETECT_VERSION_KEY in your environment and
+# *that* key will be used to get the download url from
+# artifactory.
+DETECT_VERSION_KEY=${DETECT_VERSION_KEY:-DETECT_LATEST_4}
+
+# You can specify your own download url from
+# artifactory which can bypass using the property keys
+# (this is mainly for QA purposes only)
+DETECT_SOURCE=${DETECT_SOURCE:-}
+
+# To override the default location of /tmp, specify
+# your own DETECT_JAR_PATH in your environment and
+# *that* location will be used.
+DETECT_JAR_PATH=${DETECT_JAR_PATH:-/tmp}
```

Comment:
Is it possible/allowable to have spaces in the DETECT_JAR_PATH? It's ridiculous in my opinion, but some people do do it. If DETECT_JAR_PATH has spaces, the script does throw errors:

`SPACES TEST:

export DETECT_JAR_PATH=/Users/danam/a\ b/c\ d
$ ./dana-detect.sh 
will look for : https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/4.3.0/hub-detect-4.3.0.jar
./dana-detect.sh: line 72: [: too many arguments
You have already downloaded the latest file, so the local file will be used.
running detect: java  -jar /Users/danam/a b/c d/hub-detect-4.3.0.jar 
rm: /Users/danam/a: is a directory
./dana-detect.sh: line 94: $DETECT_JAR_PATH/hub-detect-java.sh: ambiguous redirect
./dana-detect.sh: line 95: $DETECT_JAR_PATH/hub-detect-java.sh: ambiguous redirect
./dana-detect.sh: line 96: $DETECT_JAR_PATH/hub-detect-java.sh: ambiguous redirect

$ unset DETECT_JAR_PATH
$ ./dana-detect.sh 
will look for : https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/4.3.0/hub-detect-4.3.0.jar
You don't have the current file, so it will be downloaded.
getting https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/4.3.0/hub-detect-4.3.0.jar from remote
saved https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/4.3.0/hub-detect-4.3.0.jar to /tmp/hub-detect-4.3.0.jar
running detect: java  -jar /tmp/hub-detect-4.3.0.jar `

Reply:
Let's say this *is* a problem, but I'm not sure it is solvable in the scope of this PR. We should revisit this and bump it up in priority if a customer complains.